newdocs/typeclass-cheatsheet.md

```diff
@@ -0,0 +1,223 @@
+      
+  class (UpdateLedger b) => LedgerSupportsMempool b where
+    txInvariant :: GenTx b → Bool                                                -- check if internal invariants of the transaction hold
+    applyTx   :: lc → WhetherToIntervene → SlotNo → tx → tls → Except txerr (tls, Validated tx)      -- apply an unvalidated transaction
+    reapplyTx :: lc →            SlotNo → Validated tx → tls → Except txerr tls          -- apply a previously validated transaction ...
+    txsMaxBytes :: tls → Word32                                   -- max number of bytes of transactions that can be put into a block    
+    txInBlockSize :: tx → Word32                                  -- post-serialisation size in bytes of a 'GenTx b'
+    txForgetValidated :: Validated tx → tx                        -- discard the evidence that transaction has been previously validated     
+```
+
+## Quick Reference for Useful Base Types
+### ouroboros-consensus:
+
+**ouroboros-consensus/src/Ouroboros/Consensus/Forecast.hs`**
+
+``` haskell
+-- | Forecast the effect of time ticking
+data Forecast a = Forecast {
+      forecastAt  :: WithOrigin SlotNo
+
+      -- Precondition: @At s >= forecastAt@
+    , forecastFor :: SlotNo -> Except OutsideForecastRange (Ticked a)
+    }
+```
+
+**ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs**
+
+``` haskell
+-- | The result of invoke a ledger function that does validation
+data LedgerResult l a = LedgerResult
+  { lrEvents :: [AuxLedgerEvent l]
+  , lrResult :: !a
+  }
+  deriving (Foldable, Functor, Traversable)
+```
+
+### cardano-base:
+
+**slotting/src/Cardano/Slotting/Slot.hs**
+
+```haskell
+data WithOrigin t = Origin | At !t
+
+-- | The 0-based index for the Ourboros time slot.
+newtype SlotNo = SlotNo {unSlotNo :: Word64}
+```
+
+### ouroboros-network:
+
+**ouroboros-network/src/Ouroboros/Network/Block.hs**
+
+``` haskell
+data ChainHash b = GenesisHash | BlockHash !(HeaderHash b)  -- Our*.Network.Block
+
+-- | Header fields we expect to be present in a block
+--
+-- These fields are lazy because they are extracted from a block or block
+-- header; this type is not intended for storage.
+data HeaderFields b = HeaderFields {
+      headerFieldSlot    :: SlotNo
+    , headerFieldBlockNo :: BlockNo
+    , headerFieldHash    :: HeaderHash b
+      -- ^ NOTE: this field is last so that the derived 'Eq' and 'Ord'
+      -- instances first compare the slot and block numbers, which is cheaper
+      -- than comparing hashes.
+    }
+  deriving (Generic)
+
+-- | A point on the chain is identified by its 'Slot' and 'HeaderHash'.
+--
+-- The 'Slot' tells us where to look and the 'HeaderHash' either simply serves
+-- as a check, or in some contexts it disambiguates blocks from different forks
+-- that were in the same slot.
+--
+-- It's a newtype rather than a type synonym, because using a type synonym
+-- would lead to ambiguity, since HeaderHash is a non-injective type family.
+newtype Point block = Point
```

Comment:
As a data point, I think most of us Consensus members typically think of `Point` as its pattern synonyms `GenesisPoint` and `BlockPoint` instead of this defn.

Reply:
> As a data point, I think most of us Consensus members typically think of `Point` as its pattern synonyms `GenesisPoint` and `BlockPoint` instead of this defn.

I put a little text/code to that effect.